### ISSUE TITLE
levels: outdoor linedefs need to face the right way.

### ISSUE THREAD

**Issue by mc776** [freedoom/freedoom#996]
https://www.youtube.com/watch?v=9GoSRw4TDw0

To summarize the vanilla bug behaviour:
1. Have two sectors, both with sky ceilings.
2. Make one of them have a floor high enough to shoot when you're standing on the other one.
3. Set the linedef between them so that its front side is towards the "inside" of the wall and the back facing whoever's standing on the lower floor.
4. Stand on the lower floor
5. Shoot a rocket at this 2-sided linedef.
6. It disappears, having registered as having been shot into the sky.

A lot of Freedoom's maps have no accommodation for this. These lines are everywhere.

Inuk's got a script for this for Ultimate Doom Builder but I can't use UDB, hence posting this issue [EDIT: updated after some discussion on Discord]:
```
`#version 4`;
`#name Select raised walls facing inwards`;
`#description Selects any non-action linedef that faces inside a raised sector. This includes selecting walls that can cause the disappearing rockets bug.`;
UDB.Map.getLinedefs().filter(ld =>
    ld.back !== null &&
    ld.action == 0 &&
    ld.front.sector.floorHeight > ld.back.sector.floorHeight
).forEach(ld => ld.selected = true);
```

Basically we need to do this (also credit to Inuk for the diagram):
![noingrownhairs](https://github.com/freedoom/freedoom/assets/24984517/cb1ce62e-0ad2-4436-a0d4-6c569e2c213c)

**Comment by selliott512** [comment]
That's a good explanation, but I failed to reproduce it just now. In FD E2M1 I tried raising the floor height of sector #163 (in starting area) to 200. I tried flipping the linedefs between it at neighboring lower sector #164, but it did not seem to matter for either Chocolate Doom or GZDoom. Maybe there are other factors.

The script you quoted does not seem to care about the sky texture, but maybe that's ok - there should not be any harm in flipping it everywhere. The "good" orientation is the one that makes the most sense.

In lieu of UDB I wonder how hard it would be to write a Python script for FD that does what that UDB script does. That way we can run it any time we want without any external tools.

**Comment by mc776** [comment]
Reopening this so it's easy to track down for reference.
